### ISSUE TITLE
Add link to parent object

### ISSUE THREAD

**Issue by ilg-ul** [kgabis/parson#66]
Would it be possible to add a `parent` field to objects, in order to allow back-navigation in the hierarchy?

**Comment by kgabis** [comment]
Yes, it would be. I could add ```json_value_get_parent``` to get parent value in addition to ``json_object_get_enclosing_value``` and ```json_array_get_enclosing_value``` to get values that point to object/array. It would be a bit clunky, but then again: it's not something people use very often. Any thoughts on this?

**Comment by ilg-ul** [comment]
> it's not something people use very often

indeed, for simple applications it is not needed.

and for complex applications that do recursive parsing it can be overcome by always passing the root to each call.

if you think an extra pointer is too expensive for most use cases, you can add it conditionally, and enable it with  a preprocessor definition. 

or you can postpone it, until more requests are recorded.

just for the record, except the need to reach the root node during some recursions (easily overcome), using your parser is quite straightforward and convenient. thank you!

**Comment by kgabis** [comment]
I *might* add this feature in future, because it seems pretty easy to implement and it only requires 3 functions added to the api. I'll leave this issue open to not forget about it and in case someone else has a similar idea.

Anyway, it's nice to hear that you find parson useful :)

**Comment by marcomorain** [comment]
I have another use-case where parent pointers might be useful. 

I recently started using parson and I incorrectly thought I could share a JSON array as a child of two seperate JSON objects. I ended up with a crash when freeing the second object (a double free of the array I think). 

If objects have a parent pointer then parson could assert much earlier about my incorrect use of the API - when setting an item to be the child of another the item should have a null parent pointer.
